### DISCUSSION TITLE
Question regarding ReserveTagPhysIDs

### DISCUSSION THREAD

**Discussion by saisandeepdammati** [PrincetonUniversity/athena#468] [Q&A]
Howdy!

I have a question regarding tags that being reserved for physics id based on enabled physics. 

I see that if only HYDRO is enabled along with (NSCALARS > 0) in 2D, four physics ids (namely 1, 2, 3, 4) get reserved for hydro, and four physics ids (namely 5, 6, 7, 8) are reserved for NSCALARS. However, currently only cc_phys_id_ is set to value '1', cc_flx_phys_id_ is set to value '2' in HYDRO cell centered bvals followed by cc_phys_id_ = 5 and cc_flx_phys_id_ = 6 in NSCALARS cell centered bvals.

I am not sure where physics ids = 3, 4, 7, 8 are currently being used in the code? Are they currently left unused? Also, can we limit the max physics ids reserved for each of HYDRO, MAGNETIC_FIELDS, SELF_GRAVITY and NSCALARS by setting 

**static constexpr int max_phys_id** in **bvals_cc.hpp** and **bvals_fc.hpp** to lower values say **2**?

Thank you very much.

**Comment by felker** [comment]
https://github.com/PrincetonUniversity/athena/blob/e34cd25ab4fda6e2026a4039d25e365bf4b496c3/src/mesh/meshblock.cpp#L141-L167

As the comment points out, we greedily reserve 4x IDs for hydro. All 4x of them are used for SMR/AMR calculations.

Why do you want to lower the reserved IDs?
